water/lib/Dialect/Wave/IR/WaveInterfaces.cpp

```diff
@@ -27,46 +27,56 @@ LogicalResult wave::verifyWaveIndexMappings(Operation *op) {
   if (!attribute)
     return success();
 
-  auto dictAttr = dyn_cast<DictionaryAttr>(attribute);
-  if (!dictAttr)
-    return op->emitError("'index' attribute must be a dictionary");
-
-  for (auto named : dictAttr) {
-    auto val = named.getValue();
-    if (!isa<wave::WaveIndexMappingAttr>(val))
-      return op->emitError(
-                 "index attribute values must be WaveIndexMappingAttr, got ")
-             << val;
-
-    auto mapping = cast<wave::WaveIndexMappingAttr>(val);
-    auto checkNoDims = [&](AffineMap map, StringRef which) -> LogicalResult {
-      if (map.getNumDims() != 0)
-        return op->emitError(
-                   "wave indexing " + which +
-                   " map should have no dimensions, only symbols, got ")
-               << map.getNumDims() << " dimensions for symbol "
-               << named.getName();
-      return success();
-    };
-
-    AffineMap startMap = mapping.getStart();
-    AffineMap stepMap = mapping.getStep();
-    AffineMap strideMap = mapping.getStride();
-    if (failed(checkNoDims(startMap, "start")) ||
-        failed(checkNoDims(stepMap, "step")) ||
-        failed(checkNoDims(strideMap, "stride")))
-      return failure();
+  auto arr = dyn_cast<ArrayAttr>(attribute);
+  if (!arr)
+    return op->emitError("'index' attribute must be an array of dictionaries");
+  SmallVector<DictionaryAttr, 3> dicts;
```

Comment:
Nit: why 3 is hardcoded? I get that it is from the `mma` op having a max three of this, but should we care here / is there an obvious performance benefit?

Reply:
I see! Sounds good 👍